From 2bcd30e0c88ccf2a7cbe7ec8248d64a198559fcc Mon Sep 17 00:00:00 2001
From: jeremyd2019 <github@jdrake.com>
Date: Wed, 8 Sep 2021 19:08:42 -0700
Subject: [PATCH] prefix symbols with _ for 32-bit x86 Windows

In a case that I believe can only be hit for Clang i686-*-windows-gnu (AKA MinGW), symbols in asm need to be prefixed with `_`.  Fixes #237
---
 src/cdefs-compat.h | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/cdefs-compat.h b/src/cdefs-compat.h
index c7deeb1..834c547 100644
--- a/src/cdefs-compat.h
+++ b/src/cdefs-compat.h
@@ -59,14 +59,19 @@
 #endif	/* __warn_references */
 #endif	/* __STDC__ */
 #elif defined(__clang__) /* CLANG */
+#if defined(_WIN32) && defined(_X86_)
+#define openlibm_asm_symbol_prefix "_"
+#else
+#define openlibm_asm_symbol_prefix ""
+#endif
 #ifdef __STDC__
 #define openlibm_weak_reference(sym,alias)     \
-    __asm__(".weak_reference " #alias); \
-    __asm__(".set " #alias ", " #sym)
+    __asm__(".weak_reference " openlibm_asm_symbol_prefix #alias); \
+    __asm__(".set " openlibm_asm_symbol_prefix #alias ", " openlibm_asm_symbol_prefix #sym)
 #else
 #define openlibm_weak_reference(sym,alias)     \
-    __asm__(".weak_reference alias");\
-    __asm__(".set alias, sym")
+    __asm__(".weak_reference openlibm_asm_symbol_prefix/**/alias");\
+    __asm__(".set openlibm_asm_symbol_prefix/**/alias, openlibm_asm_symbol_prefix/**/sym")
 #endif
 #else	/* !__ELF__ */
 #ifdef __STDC__
